x-pack/plugins/enterprise_search/public/applications/shared/kibana/kibana_logic.test.ts

```diff
@@ -33,6 +33,12 @@ describe('KibanaLogic', () => {
       expect(KibanaLogic.values.config).toEqual({});
     });
 
+    it('gracefully handles disabled security', () => {
```

Comment:
Repost of previously exisiting comment attached to random code to allow threading

---

I think this nav item should be active on the "Add your first source" page:
![image](https://user-images.githubusercontent.com/11838280/122589617-de38ae00-d036-11eb-9bbd-7c2d1f09fdf3.png)

Reply:
We can't do this easily without some tweaking our `isActive` code or possibly adding a one-off. The problem is that the logic for highlighting parents when children are visible is very basic: if the parent is `sources/`, then any URL that starts with `sources/` highlights the parents. This is fine for `sources/add_source`... but unfortunately it also causes highlighting for `sources/{source_id}/etc...`:

<img width="271" alt="Screen Shot 2021-06-18 at 10 03 38 AM" src="https://user-images.githubusercontent.com/549407/122594884-7ecda480-d01c-11eb-826d-017945572f48.png">

Although if this issue is specific to parent links that sometimes have children and sometimes don't, it's _possible_ I can try to add some more generic logic to nav links with `items`..., but it's not gonna be great 🤷

Reply:
Hm, I think this looks quite neat, actually:
<img width="271" alt="Screen Shot 2021-06-18 at 10 03 38 AM" src="https://user-images.githubusercontent.com/549407/122594884-7ecda480-d01c-11eb-826d-017945572f48.png">

But I don't want to go against established patterns. I'm fine with leaving this as-is unless there's an easy non-hacky fix.

Reply:
I can check with with EUI if we're supposed to have multiple parent/child `isSelected` states - if so we could just turn on active for all links with child subnavs - but their docs kind of lead me to think no 🤔 

<img width="212" alt="" src="https://user-images.githubusercontent.com/549407/122598050-23ea7c00-d021-11eb-9134-8059dc978b1d.png">

Reply:
I have this working locally!

![nav](https://user-images.githubusercontent.com/549407/122600718-0e775100-d025-11eb-81e3-80d1fcb73600.gif)

But, I'm not a huge fan of putting it in this already complex PR. I'd like to open it in a separate follow-up PR if that's cool. As a bonus, I think AS could probably use this as well for its Create Engine/Meta Engine routes :)

Reply:
Awesome, thank you! Follow-up PR sounds good!